tests/CustomerTest.php

```diff
@@ -2,337 +2,220 @@
-            array(),
-            array('id' => 'card_123', 'object' => 'card')
+        $this->expectsRequest(
+            'get',
+            '/v1/customers/' . TEST_RESOURCE_ID . '/sources/' . TEST_SOURCE_ID
         );
+        $resource = Customer::retrieveSource(TEST_RESOURCE_ID, TEST_SOURCE_ID);
+        $this->assertSame("Stripe\\BankAccount", get_class($resource));
 
-        $source = Customer::retrieveSource(
-            'cus_123',
-            'card_123'
-        );
-
-        $this->assertSame('card_123', $source->id);
-        $this->assertSame('card', $source->object);
     }
 
-    public function testStaticUpdateSource()
+    public function testCanUpdateSource()
     {
-        $this->mockRequest(
-            'POST',
-            '/v1/customers/cus_123/sources/card_123',
-            array('metadata' => array('foo' => 'bar')),
-            array('id' => 'card_123', 'object' => 'card')
-        );
-
-        $source = Customer::updateSource(
-            'cus_123',
-            'card_123',
-            array('metadata' => array('foo' => 'bar'))
+        $this->expectsRequest(
+            'post',
+            '/v1/customers/' . TEST_RESOURCE_ID . '/sources/' . TEST_SOURCE_ID
         );
+        $resource = Customer::updateSource(TEST_RESOURCE_ID, TEST_SOURCE_ID, array("name" => "name"));
+        // stripe-mock returns a Card on this method and not a bank account
+        $this->assertSame("Stripe\\Card", get_class($resource));
 
-        $this->assertSame('card_123', $source->id);
-        $this->assertSame('card', $source->object);
     }
 
-    public function testStaticDeleteSource()
+    public function testCanDeleteSource()
     {
-        $this->mockRequest(
-            'DELETE',
-            '/v1/customers/cus_123/sources/card_123',
-            array(),
-            array('id' => 'card_123', 'deleted' => true)
+        $this->expectsRequest(
+            'delete',
+            '/v1/customers/' . TEST_RESOURCE_ID . '/sources/' . TEST_SOURCE_ID
         );
+        $resource = Customer::deleteSource(TEST_RESOURCE_ID, TEST_SOURCE_ID);
+        $this->assertSame("Stripe\\BankAccount", get_class($resource));
 
-        $source = Customer::deleteSource(
-            'cus_123',
-            'card_123'
-        );
-
-        $this->assertSame('card_123', $source->id);
-        $this->assertSame(true, $source->deleted);
     }
 
-    public function testStaticAllSources()
+    public function testCanListSources()
     {
-        $this->mockRequest(
-            'GET',
-            '/v1/customers/cus_123/sources',
-            array(),
-            array('object' => 'list', 'data' => array())
-        );
-
-        $sources = Customer::allsources(
-            'cus_123'
+        $this->expectsRequest(
+            'get',
+            '/v1/customers/' . TEST_RESOURCE_ID . '/sources'
         );
+        $resources = Customer::allSources(TEST_RESOURCE_ID);
+        $this->assertTrue(is_array($resources->data));
```

Comment:
this one doesn't test the class because stripe-mock currently returns a list of `Account` which is incorrect/impossible. Flagged internally

Reply:
Ack. Can you add a comment with a TODO so that hopefully we remember to add the proper assertion once stripe-mock does the right thing?

Reply:
Not sure we need an assertion here to be honest and it does not really matter since everything is dynamic and any issue would come from stripe-mock or the API and not the library

Reply:
Mostly to be consistent with other list tests, but fair enough, we don't really need to assert the contents of every list returned by stripe-mock -- we're already checking that a list is returned which is probably enough to ensure that we're hitting the correct URL.